src/ts/eventdispatcher.ts

```diff
@@ -196,28 +197,76 @@ class EventListenerWrapper<Sender, Args> {
  */
 class RateLimitedEventListenerWrapper<Sender, Args> extends EventListenerWrapper<Sender, Args> {
 
-  private rateMs: number;
-  private rateLimitingEventListener: EventListener<Sender, Args>;
+  // Timout in MS where the last seen event is delayed
+  private static NEXT_EVENT_TIMEOUT_MS = 50;
 
-  private lastFireTime: number;
+  private readonly rateMs: number;
+  private readonly rateLimitingEventListener: EventListener<Sender, Args>;
+
+  // save last seen event attributes
+  private lastSeenEvent: any;
+
+  private rateLimitTimout: Timeout;
+  private waitAndFireLastTimout: Timeout;
+
+  // Boolean to track if rateLimit timeout is currently active
+  private isRateLimiting: boolean = false;
```

Comment:
The name is very misleading, to me it signals that rate limiting can be active or disabled. It is always active though.